### PR TITLE
[Chore]Add storybook

### PR DESCRIPTION
**Pull request by nerdlet**
## Description

Fixes https://github.com/CodeForAfrica/ui/issues/9

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const path = require("path");
```
Comment (kilemensi) [inline .storybook/main.js:1]: Should be in `packages/ui`, no?

File: package.json [hunk 2]
```diff
@@ -8,9 +8,28 @@
...
-    "turbo": "latest"
+    "prop-types": "^15.8.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "storybook-addon-material-ui": "^0.9.0-alpha.24",
+    "storybook-addon-next-router": "^3.1.1",
+    "svg-url-loader": "^7.1.1",
+    "turbo": "latest",
+    "typescript": "^4.6.2",
+    "webpack": "^5.70.0"
```
Comment (kilemensi) [inline package.json:32]: Shouldn't be here as well.

File: package.json [hunk 3]
```diff
@@ -8,9 +8,28 @@
     "lint": "turbo run lint"
```
Comment (kilemensi) [inline package.json:8]:
```suggestion
    "storybook": "turbo run storybook",
    "lint": "turbo run lint"
```

File: packages/ui/package.json [hunk 4]
```diff
@@ -3,7 +3,22 @@
   "version": "0.0.0",
   "main": "./index.js",
   "license": "MIT",
+  "scripts": {
+    "build": "turbo run build",
+    "dev": "turbo run dev --parallel",
+    "lint": "turbo run lint",
+    "storybook": "start-storybook -p 9001 -c .storybook"
```
Comment (kilemensi) [inline packages/ui/package.json:10]: Isn't `.storybook` the default config dir?

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: package.json [hunk 5]
```diff
@@ -14,6 +14,10 @@
     "lint-staged": "pnpm eslint --fix --ext '.js,.json,.md,.yml'"
   },
   "devDependencies": {
+    "@babel/core": "^7.17.5",
+    "@material-ui/core": "^4.12.3",
+    "babel-loader": "^8.2.3",
+    "next": "^12.1.0",
```
Comment (kilemensi) [inline package.json:20]: Why do we need any of these at root?

File: package.json [hunk 6]
```diff
@@ -31,7 +35,11 @@
     "jscodeshift": "^0.13.0",
     "lint-staged": "^12.3.5",
     "prettier": "^2.5.1",
-    "turbo": "latest"
+    "prop-types": "^15.8.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "turbo": "latest",
+    "typescript": "^4.6.2"
```
Comment (kilemensi) [inline package.json:42]:
```suggestion
    "prop-types": "^15.8.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "turbo": "latest",
    "typescript": "^4.6.2"
```
Also these (except `turbo`)

**Review by kilemensi — COMMENTED** [review]
👍🏽

**Review by KhadijaMahanga — APPROVED** [review]
🚀 🚀

File: packages/ui/package.json [hunk 7]
```diff
@@ -5,21 +5,36 @@
...
     "@types/react": "^17.0.40",
+    "babel-loader": "^8.2.3",
     "eslint": "^8.11.0",
     "eslint-config-commons-ui": "workspace:*",
     "prettier": "^2.6.0",
+    "prop-types": "^15.8.1",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "typescript": "^4.6.2"
+    "storybook-addon-material-ui": "^0.9.0-alpha.24",
```
Comment (kilemensi) [inline packages/ui/package.json:33]: Does this work with MUI v5? I see another [package](https://www.npmjs.com/package/storybook-addon-material-ui5) forked from this just to support [MUI 5](https://github.com/alexpanov/storybook-addon-material-ui/commits/version-1)

File: packages/ui/index.stories.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
```
Comment (kilemensi) [inline packages/ui/index.stories.js:1]: If this is a story for `Button.js`, it should be in the `Button.stories.js` file, no?

File: packages/ui/.storybook/main.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    config.module.rules.push({
+      test: /\.svg$/,
+      use: [
+        "@svgr/webpack",
+        {
+          loader: "svg-url-loader",
+          options: {
+            // https://github.com/bhovhannes/svg-url-loader#iesafe
+            iesafe: true,
+          },
```
Comment (kilemensi) [inline packages/ui/.storybook/main.js:39]:
```suggestion
```
We no longer need to support IE me thinks.

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

Why are we deleting things in `apps/web`?

File: packages/ui/package.json (hunk 7 above)
Comment (nerdlet) [reply packages/ui/package.json]: @kilemensi True true...storybook-addon-material-ui isn't supported by version 5 of material ui. Updated that to show add https://www.npmjs.com/package/storybook-addon-material-ui5

File: packages/ui/Button.stories.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+import { Button } from "./index";
+
+export default {
+  title: "Component/Button",
+  argTypes: {},
+};
+
+function Template({ ...args }) {
+  return <Button {...args} />;
```
Comment (gertie-sheshe) [inline packages/ui/Button.stories.js:11]: Button does not have any args to destructure so these can be removed

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Why am I even seeing changes in `/apps`? This PR should just be about `/packages` (`ui` and _may be_ `config`)

**Comment by nerdlet** [comment]
> 👍🏽
> 
> Why am I even seeing changes in `/apps`? This PR should just be about `/packages` (`ui` and _may be_ `config`)

@kilemensi That was because i had to remove the  port info in the scripts

**Comment by kilemensi** [comment]
> > 👍🏽
> > Why am I even seeing changes in `/apps`? This PR should just be about `/packages` (`ui` and _may be_ `config`)
> 
> @kilemensi That was because i had to remove the port info in the scripts

Does port info has anything to do with Storybook @nerdlet ?

**Pull request merged**